src/main/java/github/scarsz/discordsrv/api/SlashCommandManager.java

```diff
@@ -0,0 +1,142 @@
+/*-
+ * LICENSE
+ * DiscordSRV
+ * -------------
+ * Copyright (C) 2016 - 2022 Austin "Scarsz" Shapiro
+ * -------------
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * END
+ */
+
+package github.scarsz.discordsrv.api;
+
+import github.scarsz.discordsrv.DiscordSRV;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.exceptions.RateLimitedException;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class SlashCommandManager {
+    private final Map<Plugin, PluginCommands> commandsMap = new HashMap<>();
+
+    /**
+     * Gets the Plugin Commands for a plugin
+     * @param plugin plugin to get commands of
+     * @return null if this plugin never added their commands here, or else the {@link PluginCommands}
+     */
+    public PluginCommands getCommands(Plugin plugin) {
+        return commandsMap.get(plugin);
+    }
+
+    public void reloadSlashCommands() {
+        Set<CommandData> commands = new HashSet<>();
+        for (PluginCommands value : commandsMap.values()) {
+            if (value.plugin.isEnabled()) {
+                commands.addAll(value.data);
+            }
+        }
+        if (!commands.isEmpty()) return; //no need to register if no commands will be registered
+        Map<Guild, RegistrationResult> errors = new HashMap<>();
```

Comment:
The `Map` data structure is pointless here, you aren't going to get the values by key as you're simply iterating over all the values anyway.

You could just have a `List<RegistrationResult>` where the `RegistrationResult` also holds the Guild

**Don't forget to use a synchronized list**